flips/20211007-transaction-fees.md

```diff
@@ -0,0 +1,327 @@
+- <img src="https://render.githubusercontent.com/render/math?math=F_I"> : Inclusion part of the transaction fees.
+- <img src="https://render.githubusercontent.com/render/math?math=F_E"> : Execution part of the transaction fees.
+- <img src="https://render.githubusercontent.com/render/math?math=I">: Inclusion effort; The time invariant effort required to handle this transaction (include it into a collection, check signatures are valid, check that the payer can pay, transfer the transaction over the wire,...). The inclusion effort is known before the transaction is executed.
+- <img src="https://render.githubusercontent.com/render/math?math=E">: Execution effort; The time invariant effort required to execute this transaction (run the script, process the results, send results to verification node, verify the results, ...). The Execution effort cannot be know without executing the 
+- <img src="https://render.githubusercontent.com/render/math?math=E%5E%7B%5Ctext%7Bmax%7D%7D">: Maximum execution effort (gas/computation limit).
+- <img src="https://render.githubusercontent.com/render/math?math=p_E(E)"> or <img src="https://render.githubusercontent.com/render/math?math=p_I(I)">: Flow cost of effort. In general execution and inclusion effort can be priced differently.
+
+The transaction fees equation can then be written in the following form.
+
+<img src="https://render.githubusercontent.com/render/math?math=F%20%3D%20s%20%5Cleft%5B%20F_I%20%2B%20F_E%20%5Cright%5D%20%3D%20s%20%5Cleft%5B%20p_I(I)%20%2B%20p_E(E)%20%5Cright%5D">
+
+The maximum fees a transaction can incur can be written with the equation below. 
+
+<img src="https://render.githubusercontent.com/render/math?math=F%5E%7B%5Ctext%7Bmax%7D%7D%20%3D%20s%20%5Cleft%5B%20p_I(I)%20%2B%20p_E(E%5E%7B%5Ctext%7Bmax%7D%7D)%20%5Cright%5D">
+
+### Surge factor
+
+The surge factor serves only one purpose, which is to give all transactions on the network a surcharge or potentially a discount when the network load is high or low respectively. It should be defined on a smart contract and adjustable via the service account admin resource and should be accessible for anyone to read. 
+
+The surge factor could be adjusted manually from the start, by manually moniyoring the network and sending a transaction to update the surge factor. This would mean responding to network load would be very slow. 
+
+A fist automated solution to adjusting the surge factor would be a service that would monitor the network load and send a transaction to update the surge factor when there was a change in the network load.
+
+When automation will be in place for adjusting the surge factor, the surge factor could be adjusted frequently (in the span it takes to run a few blocks and detect a surge), but it should not change to drastically otherwise users cannot respond to the change.
```

Comment:
```suggestion
When automation will be in place for adjusting the surge factor, the surge factor could be adjusted frequently (in the span it takes to run a few blocks and detect a surge), but it should not change too drastically otherwise users cannot respond to the change.
```